services/workflows-service/src/collection-flow/controllers/collection-flow.controller.ts

```diff
@@ -199,8 +199,8 @@ export class CollectionFlowController {
               entity: {
                 data: {
                   additionalInfo: {
-                    directors,
-                    ubos,
+                    directors: directors?.length ? directors : undefined,
+                    ubos: ubos?.length ? ubos : undefined,
```

Comment:
Don't have to do it if there's no need, just might be safer.
```suggestion
                    ...(directors && { directors }),
                    ...(ubos && { ubos }),
```